### ISSUE TITLE
timeframe doesn't work fully on tasks only started by IRC plugin

### ISSUE THREAD

**Issue by CrazyCanadianer** [Flexget/Flexget#2448]

**Comment by CrazyCanadianer** [comment]
I had the same thought after reading the other guy's ticket but I don't know how I would verify.
If I force run the task from CLI it seems to "catch up" and download everything that would have normally expired and downloaded automatically.

I'm unsure how to diagnose this...

**Comment by gazpachoking** [comment]
I think the best solution is to add this task to your 'schedules' this way it can consider whether the timeframe has passed and correctly get any of the backup entries it's been saving in backlog. You are probably going to get warnings about not having an input plugin, and not having any entries in the task. We should figure out how to handle this case in the future, but for now you could add the following to the task just to suppress the warnings:
```yaml
no_entries_ok: yes
mock: []
```

**Comment by CrazyCanadianer** [comment]
Dude, this has been driving me insane for over a month. I'm almost certain it used to work, but looking back I don't actually know.

Just the fact that someone else has acknowledged this is fantastic news to me.

I'll schedule the task, hopefully this takes care of the ones that are "missed" this way.

**Comment by frankyw** [comment]
Also think I'm having some issues that may be related to this.

Have similar behaviour, except the wrong quality is downloaded after the timeframe expires.

config snippet ([full config](https://github.com/frankyw/flexget/blob/master/config.yml))

`  tv-settings:
    <<: *tv_lookup
    content_size:
      max: 10000
      min: 200
    sequence:
      - configure_series:
          from:
            <<: *trakt_tv_list
          settings:
            exact: no
            target: 1080p hdtv+
            quality: 1080p webrip+
            timeframe: 18 hours
            propers: 2 days
            identified_by: ep
      - configure_series:
          from:
            <<: *trakt_tv4k_list
          settings:
            exact: no
            target: 2160p webrip+
            quality: 1080p webrip+
            timeframe: 3 days
            propers: 3 days
            identified_by: ep`

Series show:

`│ S01E05     │ 3d 16h    │ The Mandalorian S01E05 iNTERNAL 720p WEB H264-AMRAP                                               │ 720p webdl h264           │
│            │           │ The Mandalorian S01E05 Chapter 5 720p DSNP WEB-DL DDP5 1 H 264-NTb                                │ 720p webdl h264 dd+5.1    │
│            │           │ The Mandalorian S01E05 720p x265-ZMNT                                                             │ 720p h265                 │
│            │           │ The Mandalorian S01E05 1080p WEBRiP x264-PETRiFiED                                                │ 1080p webrip h264         │
│            │           │ The.Mandalorian.S01E05.1080p.WEBRiP.x264-PETRiFiED *                                              │ 1080p webrip h264         │
│            │           │ The Mandalorian S01E05 1080p HEVC x265-MeGusta                                                    │ 1080p h265                │
│            │           │ The.Mandalorian.S01E05.Chapter.5.720p.DSNP.WEB-DL.DDP5.1.H.264-NTb                                │ 720p webdl h264 dd+5.1    │
│            │           │ [REQ] The Mandalorian S01E05 HDR 2160p WEBRiP x265-PETRiFiED                                      │ 2160p webrip h265         │
│            │           │ The Mandalorian S01E05 HDR 2160p WEBRiP x265-PETRiFiED                                            │ 2160p webrip h265         │
│            │           │ The Mandalorian S01E05 2160p WEBRip X264 Ac3 SNAKE                                                │ 2160p webrip h264 ac3     │
│            │           │ The.Mandalorian.S01E05.HDR.2160p.WEBRiP.x265-PETRiFiED                                            │ 2160p webrip h265`

Log snippet:

`2019-12-09 03:53 INFO     task          search-tv       Plugin next_series_episodes has requested task to be ran again after execution has completed. Reason: Look for next season
2019-12-09 03:54 INFO     flexget.components.series.series search-tv       `The Mandalorian (2019)`: timeframe waiting for 05h:45min. Currently best is `The.Mandalorian.S01E05.1080p.WEBRiP.x264-PETRiFiED`.
2019-12-09 03:54 INFO     task          search-tv       Rerunning the task in case better resolution can be achieved.
2019-12-09 03:54 INFO     flexget.components.series.series search-tv       `The Mandalorian (2019)`: timeframe waiting for 05h:44min. Currently best is `The.Mandalorian.S01E05.1080p.WEBRiP.x264-PETRiFiED`.
2019-12-09 09:53 INFO     task          search-tv       Plugin next_series_episodes has requested task to be ran again after execution has completed. Reason: Look for next season
2019-12-09 09:54 INFO     flexget.components.series.series search-tv       Timeframe expired, releasing quality restriction.
2019-12-09 09:54 INFO     download      search-tv       Downloading: The.Mandalorian.S01E05.1080p.WEBRiP.x264-PETRiFiED
2019-12-09 09:54 INFO     transmission  search-tv       "The.Mandalorian.S01E05.1080p.WEBRiP.x264-PETRiFiED" torrent added to transmission`

**Comment by paranoidi** [comment]
@frankyw very interesting use of sequence and configure series, I am not sure what the end result of that will be. Later use may just override former in weird ways ..

**Comment by gazpachoking** [comment]
I've used `configure_series` like that inside sequence, it should work fine (unless you have some overlapping shows between those 2 lists, in which case I can't be sure what will happen.) @frankyw What do you expect it should have grabbed instead, is that show on the 4k list? It seems the non-irc task is the one that grabbed after timeframe expired, which probably means your issue is unrelated.

**Comment by frankyw** [comment]
@gazpachoking No overlapping shows between lists. I was expecting a 2160p WEBRip to be grabbed, as I had defined target to be 2160p and multiple 2160p episodes had been seen inside the timeframe.

The log says the best available was 1080p Webrip, but actually the best available was the 2160p Webrip if you look at the output of the 'series show'. It basically picked something that was not the best available.

I ended up [removing](https://github.com/frankyw/flexget/commit/07ab59cab982724185200770d4ebd1d542d4a57a) target altogether and just setting quality to 2160p on the 4K list.
